File: module-ui/src/main/java/taboolib/module/ui/stored/Action.java
package taboolib.module.ui.stored;

import taboolib.common.Isolated;
import taboolib.module.ui.ClickEvent;
import org.bukkit.inventory.ItemStack;

@Isolated
public abstract class Action {

    public abstract ItemStack getCurrent(ClickEvent e);

    public abstract void setCurrent(ClickEvent e, ItemStack item);

    public abstract int getCurrentSlot(ClickEvent e);
}

File: module-ui/src/main/kotlin/taboolib/module/ui/type/Action.kt
package taboolib.module.ui.type

import org.bukkit.inventory.ItemStack
import taboolib.common.Isolated
import taboolib.module.ui.ClickEvent

@Isolated
abstract class Action {

    abstract fun getCursor(e: ClickEvent): ItemStack?

    abstract fun setCursor(e: ClickEvent, item: ItemStack?)

    abstract fun getCurrentSlot(e: ClickEvent): Int
}